### ISSUE TITLE
[Feature Request]: data support for scripts in custom launchers

### ISSUE THREAD

**Issue by manonstreet** [PRProd/HA-Firemote#293]
### Feature Request

When working with a custom launcher, the yaml config supports both script and service.  When calling service, the data key is passed to the service; however when using script, data is ignored.  Scripts support [variables](https://www.home-assistant.io/integrations/script/#passing-variables-to-scripts) in 2 different ways.  One way is via `service: script.turn_on`, which works fine since data is passed in this case.  The other option is via calling the script as a service, which I assume is what the script key is doing for firemote.  In the latter case, variables are passed as arbitrary KV pairs via data.  This currently does not work since data is not passed.

TLDR:  Pass data key for both service and script calls.

**Comment by PRProd** [comment]
@manonstreet ~ Thanks for granting my request and repeating yourself by posting over here too.  :smile:

**Comment by PRProd** [comment]
@manonstreet ~

Sorry for the lack of communication on my part.  I haven't forgotten about this, just been busy with other things.  I still want to implement your ideas.  Thanks for your patience and understanding.

-Doug

**Comment by PRProd** [comment]
Good news @manonstreet!  
CC: @ronfarbernewman

I just published a beta version [v4.0.4b1]() that I'd like for you to download and test.  I'm 95% sure I did it right.  😄  This is how I tested:
<br>

Firemote YAML snippet
```yaml
custom_launchers:
  - friendly_name: flash the light
    label: FLASH
    image_path: >-
      https://upload.wikimedia.org/wikipedia/commons/2/28/Flash-outlined-thin-circular-button.svg
    color: red
    background: white
    script: flashthelablight_with_vars
    data:
      color: green
app_launch_1: customlauncher flash the light
```
<br>

script **flashthelablight_with_vars**
```yaml
flashthelablight_with_vars:
  fields:
    color:
      selector:
        text: {}
      default: white
      required: true
  alias: FlashTheLabLight With Vars
  sequence:
  - target:
      entity_id: light.ambient_light
    action: light.toggle
    data:
      color_name: '{{ color }}'
  - delay:
      hours: 0
      minutes: 0
      seconds: 0
      milliseconds: 500
  - data: {}
    target:
      entity_id: light.ambient_light
    action: light.toggle
  mode: single
  icon: mdi:flash
  description: ''
```

This worked for me! I saw a flash of green, not white!  Then I went crazy typing in various color names because I was happy it was working 😆

That being said, it's really only the 2nd day I've tried to use fields in a script (where have I been!?!) so I'm hoping that if it breaks in a more complex example, one of you will tell me about it?  Please?

Hope to hear back from you soon!  Thanks for your patience!

-Doug

**Comment by manonstreet** [comment]
I did some quick testing, and the feature appears to work for custom_launchers actions; however it does not seem to work for button_override functions.

Appreciate your efforts on this @PRProd!

**Comment by PRProd** [comment]
@manonstreet ~

Thanks for testing!  So... technically, I did exactly what you wanted, right?  😄  Have you ever heard the saying "Computers are dumb.  They only do exactly what you tell them to do."  Well, sometimes programmers are like that too, I had the dumb.

I'll make that change and publish another beta... or I'll just send it.  Either way, thank you for your quick communication, and for your testing!  I'll let you know!

-Doug

**Comment by PRProd** [comment]
@manonstreet & @ronfarbernewman ~

All done!  I have just published Firemote version [v4.0.4](https://github.com/PRProd/HA-Firemote/releases/tag/v4.0.4) which contains this enhancement request.  Enjoy!

-Doug
<hr>

After updating to the newest version, if you are unable to see the new updates on your Firemote card, visit [this link](https://github.com/PRProd/HA-Firemote/wiki/Force-a-Refresh) for help.

**Comment by manonstreet** [comment]
Updated 4.0.4 and confirmed working for both custom launchers and button overrides.  Thanks so much 🙏.

**Comment by PRProd** [comment]
Yay!  That's great!

I'm happy ro help!  Sorry it took so long
